Gateway/Validator/CheckoutResponseValidator.php

```diff
@@ -95,23 +95,59 @@ public function validate(array $validationSubject)
 					}
 					break;
 				case "RedirectShopper":
-					$payment->setAdditionalInformation('3dActive', true);
-
-					if (!empty($paReq = $response['redirect']['data']['PaReq']) &&
-						!empty($md = $response['redirect']['data']['MD']) &&
-						!empty($issuerUrl = $response['redirect']['url']) &&
-						!empty($paymentData = $response['paymentData'])
-					) {
-						$payment->setAdditionalInformation('issuerUrl', $issuerUrl);
-						$payment->setAdditionalInformation('paRequest', $paReq);
-						$payment->setAdditionalInformation('md', $md);
-						$payment->setAdditionalInformation('paymentData', $paymentData);
+
+					$redirectUrl = null;
+					$paymentData = null;
+
+					if (isset($response['redirect']['url']) && !empty($response['redirect']['url'])) {
+						$redirectUrl = $response['redirect']['url'];
+					}
+
+					if (!empty($response['paymentData'])) {
+						$paymentData = $response['paymentData'];
+					}
+
+					// If the redirect data is there then the payment is a card payment with 3d secure
+					if (isset($response['redirect']['data'])) {
```

Comment:
Shall we be more specific here. Check if $response['redirect']['data']['PaReq'] && $response['redirect']['data']['MD']) are set move to this logic otherwise go into alternative route.